### PR TITLE
add heat transport test case

### PR DESCRIPTION
**Pull request by grubbymoon**
compared with analytical solutions

### CODE REVIEW

**Comment by endJunction** [comment]
I'd recommend to use vtu format for the input meshes.

File: Parabolic/T/1D_dirichlet/line_60_heat.prj [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+       <timesteps>
+         <pair><repeat>100</repeat><delta_t>390625</delta_t></pair>
+       </timesteps>
+   </time_stepping>
+
+
+    <output>
+        <type>VTK</type>
+        <prefix>line_60_heat</prefix>
+    </output>
```
Comment (endJunction) [inline Parabolic/T/1D_dirichlet/line_60_heat.prj]: There are 100 output files; are all of them checked or only few? If there are only few, please limit the output to those files only.

File: Parabolic/T/1D_dirichlet/line_60_heat.prj [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            </petsc>
+        </linear_solver>
+    </linear_solvers>
+
+    <time_stepping>
+     <type>FixedTimeStepping</type>
+       <t_initial> 0.0 </t_initial>
+       <t_end> 39062500 </t_end>
+       <timesteps>
+         <pair><repeat>100</repeat><delta_t>390625</delta_t></pair>
```
Comment (endJunction) [inline Parabolic/T/1D_dirichlet/line_60_heat.prj]: Discussion on delta_t: Looking on the result the temperature has changed significantly for first three elements in the first time step. Reducing current delta_t by factor 3 or so will result in a change on the first element only. On the other hand the time scale (delta_x^2_rho_Cp / K) is ~ 10^7 which is much larger than the current one will result in a smoother solution from the first time step.

File: Parabolic/T/1D_dirichlet/temperature_analytical.vtu [hunk 3]
```diff
@@ -0,0 +1,105 @@
+<?xml version="1.0"?>
+<VTKFile type="UnstructuredGrid" version="0.1" byte_order="LittleEndian" header_type="UInt32" compressor="vtkZLibDataCompressor">
+  <UnstructuredGrid>
+    <Piece NumberOfPoints="61" NumberOfCells="60">
+      <PointData>
+         <DataArray type="Float64" Name="Temperature_Analytical_2months" format="ascii" RangeMin="0" RangeMax="1">
+      1.00000000e+00   7.81511190e-01   5.79099555e-01   4.05380325e-01
```
Comment (endJunction) [inline Parabolic/T/1D_dirichlet/temperature_analytical.vtu]: The accuracy of the numbers here is around 8 to 9 digits; Better to use full precision, especially if this an analytical solution.

File: Parabolic/T/1D_neumann/line_60_heat.prj [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+       <t_end> 39062500 </t_end>
+       <timesteps>
+         <pair><repeat>100</repeat><delta_t>390625</delta_t></pair>
+       </timesteps>
+   </time_stepping>
+
+
+    <output>
+        <type>VTK</type>
+        <prefix>line_60_heat</prefix>
```
Comment (endJunction) [inline Parabolic/T/1D_neumann/line_60_heat.prj]: ~~The output file names must be different because ctest running parallel.~~
Never mind, they are in different folders. But to reduce confusion a distinctive name would be helpful.

File: Parabolic/T/1D_neumann/line_60_heat.prj [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+            </eigen>
+            <petsc>
+                <prefix>gw</prefix>
+                <parameters>-gw_ksp_type cg -gw_pc_type bjacobi -gw_ksp_rtol 1e-16 -gw_ksp_max_it 10000</parameters>
+            </petsc>
+        </linear_solver>
+    </linear_solvers>
+
+    <time_stepping>
+     <type>FixedTimeStepping</type>
```
Comment (endJunction) [inline Parabolic/T/1D_neumann/line_60_heat.prj]: indentation...

**Comment by endJunction** [comment]
Also please remove unused files...

**Comment by endJunction** [comment]
And in the Neumann conduction test the boundary condition is not fulfilled at any (computed) time judging on the `heat_flux_x` which is in [-1.03, -1.91] instead of given value (-2).

@chleh Does it depend on the wrong (wrt time step) calculation of the secondary variable?

**Comment by chleh** [comment]
> Does it depend on the wrong (wrt time step) calculation of the secondary variable?

The wrong calculation makes the heat flux output lag one time step behind. But since the Neumann BC is -2 at all times, that shouldn't have any effect at the boundary. Am I right?

**Comment by grubbymoon** [comment]
@endJunction The calculation of secondary variable should be correct. The reason of different heat flux from Neumann boundary value is the space discretization. When I tested the case with finer mesh, the values became much closer.

File: Parabolic/T/1D_dirichlet/line_60_heat.prj (hunk 1 above)
Comment (grubbymoon) [reply Parabolic/T/1D_dirichlet/line_60_heat.prj]: I have limited the output only to the checked files

**Comment by grubbymoon** [comment]
@endJunction I have changed the mesh file into vtu file

File: Parabolic/T/1D_dirichlet/line_60_heat.prj (hunk 2 above)
Comment (grubbymoon) [reply Parabolic/T/1D_dirichlet/line_60_heat.prj]: I have decrease the delta_t by factor 5, 25 and 125. The relative norm compared to analytical solution decrease from 1.6e-4 to 5.27e-5 and become stable around 4.72e-5. I would say that the reducing of the time step can result in a smoother solution but is not a key issue for this problem.

File: Parabolic/T/1D_dirichlet/temperature_analytical.vtu (hunk 3 above)
Comment (grubbymoon) [reply Parabolic/T/1D_dirichlet/temperature_analytical.vtu]: Because of the relative error between numerical solution and analytical solution is around 1e-5. 8 to 9 digits is very enough for the comparison.

File: Parabolic/T/1D_neumann/line_60_heat.prj [hunk 6]
```diff
@@ -0,0 +1,132 @@
+<?xml version="1.0" encoding="ISO-8859-1"?>
+<OpenGeoSysProject>
+    <mesh>line_60_heat.vtu</mesh>
+    <geometry>line_60_heat.gml</geometry>
+
+    <processes>
+        <process>
+            <name>GW23</name>
```
Comment (chleh) [inline Parabolic/T/1D_neumann/line_60_heat.prj]: Maybe change the name to something like heat_conduction.

File: Parabolic/T/1D_neumann/line_60_heat.prj~ [hunk 7]
```diff
@@ -0,0 +1,132 @@
+<?xml version="1.0" encoding="ISO-8859-1"?>
```
Comment (chleh) [inline Parabolic/T/1D_neumann/line_60_heat.prj~]: This seems to be a backup file telling from the `.prj~` extension.

File: Parabolic/TES/1D/line_60.msh~ [hunk 8]
```diff
@@ -0,0 +1,209 @@
+#FEM_MSH
```
Comment (chleh) [inline Parabolic/TES/1D/line_60.msh~:1]: Please remove this backup file.

**Comment by chleh** [comment]
Did you already update the data files? There is no <timeloop> config

File: Parabolic/TES/1D/line_60.msh~ (hunk 8 above)
Comment (grubbymoon) [reply Parabolic/TES/1D/line_60.msh~:1]: backup files are removed

File: Parabolic/T/1D_neumann/line_60_heat.prj~ (hunk 7 above)
Comment (grubbymoon) [reply Parabolic/T/1D_neumann/line_60_heat.prj~]: delete the .prj~ files

File: Parabolic/T/1D_neumann/line_60_heat.prj (hunk 6 above)
Comment (grubbymoon) [reply Parabolic/T/1D_neumann/line_60_heat.prj]: corrected

**Comment by endJunction** [comment]
When preparing a pull request, please make sure you don't add any backup files or some unused files.
Especially after multiple request from the reviewers to remove them.

For the sha 19f16eb there are Elliptic and TES directories touched for no reasons.

**Comment by endJunction** [comment]
Merged in [master](https://github.com/ufz/ogs-data/commit/117219b7e7a6ae337257bb50c49730cdc818b26b) (after removing unrelated stuff).
